rust-runtime/aws-smithy-http-server/src/rejection.rs

```diff
@@ -248,3 +248,6 @@ convert_to_request_rejection!(std::str::Utf8Error, InvalidUtf8);
 // tests use `[crate::body::Body]` as their body type when constructing requests (and almost
 // everyone will run a Hyper-based server in their services).
 convert_to_request_rejection!(hyper::Error, HttpBody);
+
+// In order to use `Router<lambda_http::Body>` this line works around [Simplify conversion from HTTP body error type to RequestRejection](https://github.com/awslabs/smithy-rs/issues/1364)
+convert_to_request_rejection!(Box<dyn std::error::Error + Send + Sync>, HttpBody);
```

Comment:
```suggestion
convert_to_request_rejection!(lambda_http::Error, HttpBody);
```

You want to refer to the type declaration in the dependent crate right? The dependency on `lambda_http` in `Cargo.toml` is not used otherwise.